### ISSUE TITLE
[Feature] PolKit agent

### ISSUE THREAD

**Issue by Cu3PO42** [end-4/dots-hyprland#516]
Let me preface this by saying this is not a feature request in the sense that I'm asking you to please do this now, I just want to document the idea and possibly start a discussion.

**What would you like to be added?**

I'd like to propose adding a PolKit agent to your AGS config itself. I understand you currently bundle the agent used by prior versions of Gnome, but I do believe it should be possible to build a better-looking one that integrates with the rest of the aesthetic.

**How will it help**

Beyond a visual uplift, it should be possible to get handling of fingerprint authentication much more correct than it currently is in Gnome Polkit agent.

**Extra info**

I have code for the business logic of implementing a PolKit agent in AGS on hand that I'm happy to help integrate. As for the UI, I have some ideas, but nothing concrete. The agent of modern Gnome Shell looks fairly good: 

![Gnome Polkit Agent](https://blogs.gnome.org/mclasen/files/2018/12/Screenshot-from-2018-12-01-20-53-39.png)

I also like 1Password's authorization dialog:

![image](https://github.com/end-4/dots-hyprland/assets/5256300/c45f3d8c-05d2-4b71-a0a9-6e3b2b88e40a)

Maybe something similar, where instead of the 1Password icon we add the icon of the distribution.

**Comment by end-4** [comment]
yeah it would be nice to have the polkit thing show up in a layershell since i want it to hog focus like in gnome
thanks for reminding me that it's now possible
but maybe after i make a lock screen

as for fingerprint authentication, my device doesn't have a reader so i can't test that

**Comment by b4shful** [comment]
This would be super cool! There are quite a few instances where I've been on Workspace 3 or something and not noticed a polkit/keyring dialog as they always seem to go to Workspace 1 haha. Usually ends in me rage quitting a program thinking it's not working, and later when I go back to workspace 1 I see the dialog and I'm like "oh ffs there's the issue" lol

**Comment by Cu3PO42** [comment]
You can probably work around some of these issues, with the agent showing up on the wrong workspace or force it to take focus with some configuration of your window manager. E.g. on Hyprland you could force it to `pinned` and `stayfocused`.

**Comment by shawny43** [comment]
Honestly, it's a good concept but we currently have an issue because peoples who have KDE instead of Gnome in their computer have issue with keyrings at the moment with end-4 hyprland. It's needs to be fixed asap, because Vivaldi and other apps who run flatpak doesn't recognize the keys from KDE Session ported to Hyprland.

**Comment by end-4** [comment]
oh i realized i can close this, it's already there
